src/main/kotlin/com/trackforever/controllers/ProjectController.kt

```diff
@@ -96,58 +103,132 @@ class ProjectController (@Autowired private var projectRepository: ProjectReposi
+     *                     HTTP 410 GONE if none of the projects could be found.
      */
     @CrossOrigin
     @PutMapping("/issues")
     fun setIssues(@RequestBody issues: String): ResponseEntity<Map<ProjectId, List<TrackForeverIssue>>> {
+        // Check with the prevHash
+        // prevHash matches => update that issue for that project
+        // prevHash doesn't match => add that issue to the project
+        // Duplicate issue => ignore?
         logger.debug("set issues :: ($issues)")
-        val failedProjects: MutableMap<ProjectId, List<TrackForeverIssue>> = mutableMapOf()
+        val successes: MutableMap<ProjectId, MutableList<TrackForeverIssue>> = mutableMapOf()
         val mapper = jacksonObjectMapper()
         val content: Map<ProjectId, List<TrackForeverIssue>> = mapper.readValue(issues)
         content.keys.forEach {
-            val tFProj = projectRepository.findById(it)
+            val projectId = it
+            val tFProj = projectRepository.findById(projectId)
             if (tFProj.isPresent) { // The projectKey does exist
-                content[it]?.forEach { // For non-null Lists
-                    tFProj.get().issues[it.id] = it
-                }
-                projectRepository.save(tFProj.get()) // Update the entry in the database
-            } else { // The projectKey doesn't exist in the database, then add it to the failed list with an empty list as its value
-                val failedIssues = content[it]
-                if (failedIssues == null) {
-                    failedProjects[it] = emptyList()
-                } else {
-                    failedProjects[it] = failedIssues
+                successes.putIfAbsent(it, mutableListOf())
+                val currentProject = tFProj.get()
+                content[it]?.forEach { // Go through each issue in the non-null list
+                    // Check the prevHash of the current issue
+                    // prevHash matches with hash of issue on server => update that issue
+                    // prevHash doesn't match with any => add issue
+                    val currentIssue = it
+                    lateinit var matchingIssue: TrackForeverIssue // Get the matching issue if there is one
+                    currentProject.issues.values.forEach { // Go through each issue and check prevHash matches
+                        if (it.hash == currentIssue.prevHash) { // there is a match
+                            matchingIssue = it // TODO: some optimization here
+                        }
+                    }
+                    try { // check if the match exists
+                        logger.debug("Preparing to update ${matchingIssue.id} from ${currentProject.id}")
+                        currentProject.issues.remove(matchingIssue.id)
+                    } catch (e: UninitializedPropertyAccessException) { // matchingIssue was never initialized
+                        // No need to delete anything.
+                    }
```

Comment:
Try using the try catch expression here as you don't do anything with the catch

Reply:
I'm a bit confused. Is there a nicer way of doing this in Kotlin?

Reply:
You can single-line this and I don't think you need a catch.

Reply:
I tried removing the catch, but it still throws an error. I think I'll just single-line the thing.